### PR TITLE
UnifiedPicker accessibility fixes

### PR DESCRIPTION
**Pull request by elisabethcvs**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Fixing three errors given by the edge accessibility tooling

### CODE REVIEW

File: packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx [hunk 1]
```diff
@@ -242,9 +239,13 @@ export const UnifiedPicker = <T extends {}>(props: IUnifiedPickerProps<T>): JSX.
...
                   /* eslint-enable react/jsx-no-bind */
-                  aria-activedescendant={activeDescendant}
-                  aria-owns={isExpanded ? 'suggestion-list' : undefined}
-                  aria-expanded={isExpanded}
+                  aria-activedescendant={
+                    isSuggestionsShown && focusItemIndex >= 0
+                      ? 'FloatingSuggestionsItemId-' + focusItemIndex
+                      : undefined
+                  }
+                  aria-owns={isSuggestionsShown ? 'suggestion-list' : undefined}
```
Comment (philipzloh) [inline packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx:247]: Somewhere in here I think we also need `aria-autocomplete='list'`

File: packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx [hunk 2]
```diff
@@ -242,9 +239,13 @@ export const UnifiedPicker = <T extends {}>(props: IUnifiedPickerProps<T>): JSX.
...
+                  aria-expanded={isSuggestionsShown}
                   aria-haspopup="true"
                   role="combobox"
```
Comment (philipzloh) [inline packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx:250]: Aria 1.1 guidance is to have `role='combobox'` on a container and then have the typable `<input>` retain its textbox role

e.g., https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html

File: packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx [hunk 3]
```diff
@@ -242,9 +239,13 @@ export const UnifiedPicker = <T extends {}>(props: IUnifiedPickerProps<T>): JSX.
...
-                  aria-owns={isExpanded ? 'suggestion-list' : undefined}
-                  aria-expanded={isExpanded}
+                  aria-activedescendant={
+                    isSuggestionsShown && focusItemIndex >= 0
+                      ? 'FloatingSuggestionsItemId-' + focusItemIndex
+                      : undefined
+                  }
+                  aria-owns={isSuggestionsShown ? 'suggestion-list' : undefined}
+                  aria-expanded={isSuggestionsShown}
                   aria-haspopup="true"
```
Comment (philipzloh) [inline packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx:249]: `aria-haspopup="listbox"` is preferred it seems

File: packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx (hunk 1 above)
Comment (elisabethcvs) [reply packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx:234]: Updated. Autofill was setting that value to both, but I believe you're correct and it should be list here

File: packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx (hunk 2 above)
Comment (elisabethcvs) [reply packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx:237]: Fixed in the new iteration

File: packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx (hunk 3 above)
Comment (elisabethcvs) [reply packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx]: Updated

**Comment by nebhatna** [comment]
The changes look good to me. I would get another set of eyes on this since I am new to accessibility as well :)

File: packages/experiments/src/components/UnifiedPicker/__snapshots__/UnifiedPicker.test.tsx.snap [hunk 4]
```diff
@@ -57,50 +57,53 @@ exports[`UnifiedPicker renders correctly with no items 1`] = `
...
-            onCompositionEnd={[Function]}
-            onCompositionStart={[Function]}
-            onCompositionUpdate={[Function]}
-            onFocus={[Function]}
-            onInput={[Function]}
-            onKeyDown={[Function]}
-            onPaste={[Function]}
-            role="combobox"
-            value=""
+          <div
```
Comment (philipzloh) [inline packages/experiments/src/components/UnifiedPicker/__snapshots__/UnifiedPicker.test.tsx.snap:61]: curious what this `<div>` is doing here now..

Comment (philipzloh) [reply packages/experiments/src/components/UnifiedPicker/__snapshots__/UnifiedPicker.test.tsx.snap:61]: I know it's a test snap, but is it a hint that the code generated poor HTML?

Comment (philipzloh) [reply packages/experiments/src/components/UnifiedPicker/__snapshots__/UnifiedPicker.test.tsx.snap:61]: ah, the original had
```
<div role="list">
    <input role="combobox" />
</div>
```
but looks like this change turned it into 
```
<div>
    <div role="list" />
    <div role="combobox">
        <input />
    </div>
</div>
```
probably the empty `<div role="list" />` shouldn't be there...

Comment (elisabethcvs) [reply packages/experiments/src/components/UnifiedPicker/__snapshots__/UnifiedPicker.test.tsx.snap:61]: There was an empty div if the list had no items.  Not ideal. Fixed in the next update

**Pull request merged**